afm/server.py

```diff
@@ -26,7 +26,11 @@ def __init__(self, config_path: str, port: int, *args, **kwargs):
 
     def _get_dataset(self, asset):
         # FIXME(roee88): bypass https://issues.apache.org/jira/browse/ARROW-7867
-        data_files = [f.path for f in asset.filesystem.get_file_info(FileSelector(asset.path, allow_not_found=True, recursive=True)) if f.size]
+        selector = FileSelector(asset.path, allow_not_found=True, recursive=True)
+        try:
+            data_files = [f.path for f in asset.filesystem.get_file_info(selector) if f.size]
+        except NotADirectoryError:
```

Comment:
Since it's a best effort to bypass ARROW-7867, we may want to just catch any type of error (I know, we normally shouldn't, but there are too many edge cases).